service-users-server/src/main/java/io/pestakit/users/api/endpoints/UserApiController.java

```diff
@@ -0,0 +1,114 @@
+package io.pestakit.users.api.endpoints;
+
+import io.pestakit.users.api.UsersApi;
+import io.pestakit.users.api.model.User;
+import io.pestakit.users.entities.UserEntity;
+import io.pestakit.users.repositories.UserRepository;
+import io.swagger.annotations.ApiParam;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+@Controller
+public class UserApiController implements UsersApi {
+
+
+    @Autowired
+    UserRepository userRepository;
+
+
+    @Override
+    public ResponseEntity<Object> createUser(@ApiParam(value = "", required = true) @Valid @RequestBody User user) {
+
+        if (user.getUsername().indexOf('/') != -1 || user.getUsername().contains("ERROR")) {
+            return ResponseEntity.status(422).build();
+        }
+
+        for (UserEntity userEntity : userRepository.findAll()) {
```

Comment:
Not good: what happens when we have 100'000 users in the DB? Either 1) do a findByUsername or 2) ask the DB to save and catch the JDBC error.

Reply:
We made 1)